packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx

```diff
@@ -330,36 +397,66 @@ export const useTerraform = ({ targetDir, synthCommand }: UseTerraformInput) =>
     return state
   }
 
-  const deploy = (plan?: TerraformPlan) => {
+  const deploy = () => {
     React.useEffect(() => {
       const invoke = async () => {
-        if (plan) {
-          await execTerraformApply(plan)
-        } else {
-          throw new Error("expected plan to be present but was undefined")
-        }
+        await execTerraformSynth()
```

Comment:
Given how slow synth is, adding this for every deploy seems like it will have a big impact.

Reply:
We did synth before on every action. This was just moved into the actions explicitly, due to the dependency on the generated stack in order to figure out if we should use the CLI or not. In general, I'm not a 100% happy with the state of the this entire file. I hope this can be drastically simplified not too far away.

For the slow synth, I think we'd want to look into caching the generated stacks. However, that's for another day :)